### DISCUSSION TITLE
Unsure how to fix "invalid AssertStmt" error

### DISCUSSION THREAD

**Discussion by wisheth** [dafny-lang/dafny#1843] [Q&A]
I'm getting an "invalid AssertStmt" error on parse, which is highlighted in VS Code on the line below `assert u == Factorial(r)`. `u` is defined as an int in the function and the output of `Factorial(r)` should also be an int for input int `r`, so I'm unsure why the error exists. Is there a necessary invariant somewhere else I need to set, or something I'm just not noticing?

**Comment by cpitclaudel** [comment]
Looks like a parse error (so nothing about types or missing invariants).  Can you minimize your file and share the result?
